### PR TITLE
Beautiful presentation animation

### PR DESCRIPTION
**Pull request by WenchaoD**
This option make the animation as smooth as the system 'keyboard' presentation. You can see it while comparing it with 'Ease In'.

### CODE REVIEW

File: Pickers/SWActionSheet.m [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 static const float duration = .25f;
 
-static const enum UIViewAnimationOptions options = UIViewAnimationOptionCurveEaseIn;
+static const enum UIViewAnimationOptions options = 7 << 16;
```
Comment (skywinder) [inline Pickers/SWActionSheet.m]: Where it comes from this magic constant? Can you present it as combination of `UIViewAnimationOptions` ?

Comment (WenchaoIOS) [inline Pickers/SWActionSheet.m]: When you add observer to `UIKeyboardWillShowNotification` or `UIKeyboardWillChangeFrameNotification`, you can get a `UIViewAnimationCurve` value by 
```objective-c
[notification.userInfo[UIKeyboardAnimationCurveUserInfoKey] integerValue]
```

When I print the value, it give me a `7`. But there is no UIViewAnimationCurve with a rawValue 7.
Even in swift, you have to force cast it with unsafeBitCase() function.
I use `7` in:
```objective-c
    [UIView beginAnimations:nil context:nil];
    [UIView setAnimationDuration:[notification.userInfo[UIKeyboardAnimationCurveUserInfoKey] doubleValue]];
    [UIView setAnimationCurve:(UIViewAnimationCurve)[notification.userInfo[UIKeyboardAnimationCurveUserInfoKey] integerValue]];
    _textField.bottom = [notification.userInfo[UIKeyboardFrameEndUserInfoKey] CGRectValue].origin.y;
    [UIView commitAnimations];
```
It works!

But there is difference between UIViewAnimationCurve and UIViewAnimationOptions

```objective-c
    UIViewAnimationOptionCurveEaseInOut            = 0 << 16, // default
    UIViewAnimationOptionCurveEaseIn               = 1 << 16,
    UIViewAnimationOptionCurveEaseOut              = 2 << 16,
    UIViewAnimationOptionCurveLinear               = 3 << 16,
```

and this is `UIViewAnimationCurve`

```objective-c
typedef NS_ENUM(NSInteger, UIViewAnimationCurve) {
    UIViewAnimationCurveEaseInOut,         // slow at beginning and end
    UIViewAnimationCurveEaseIn,            // slow at beginning
    UIViewAnimationCurveEaseOut,           // slow at end
    UIViewAnimationCurveLinear
};
```

As a result: 
UIViewAnimationOptionCurveEaseInOut = UIViewAnimationCurveEaseInOut << 16,
UIViewAnimationOptionCurveEaseIn = UIViewAnimationCurveEaseIn << 16
.....
and so on.

So I guess, if I use UIViewAnimationOptions, I can use 7 << 16.
It turns out it totally worked!

**Comment by skywinder** [comment]
@WenchaoIOS Ok, thank you for your update. I will take a look, how it works soon. But, tell the truth, I'm not sure, that using of undocumend constants is a good idea. Probably it's better to implement as option, then everyone could decide to use it or not.

**Comment by WenchaoIOS** [comment]
Yes, it will never fail you.

**Comment by skywinder** [comment]
@WenchaoD Hi again! I don't want to make it default animation. But I'm ready to merge PR, if you make it as option.

**Comment by WenchaoD** [comment]
OK. I'll open a new property to modify the animation options.

**Comment by Stunner** [comment]
@WenchaoD not happening I take it? 😞

**Comment by skywinder** [comment]
Closing it for a while.
I'd be more than happy to merge a PR adding this feature as soon as it will be ready, it seems like it would be useful for sure 👍

**Pull request closed without merging**